p2p/node.go

```diff
@@ -71,13 +71,18 @@ type Node struct {
 	routingDiscovery discovery.Discovery
 	pubsub           *pubsub.PubSub
 	sub              *pubsub.Subscription
+	topics           []string
 }
 
 // Config contains configuration options for a Node.
 type Config struct {
-	// Topic is a unique string representing the pubsub topic. Only Nodes which
-	// have the same topic will share messages with one another.
-	Topic string
+	// DefaultTopic is the topic upon which all orders are broadcast. This is the
+	// default topic nodes share on if a custom topic is not specified at startup
```

Comment:
I think it makes more sense in the `p2p` package to separate these into `PublishTopics` and `SubscribeTopics`. We will publish all messages to each topic in `PublishTopics` and subscribe to each topic in `SubscribeTopics`. IMO it's a better separation of concerns (the p2p package doesn't need to know about the difference between default topics and user-defined topics) and also offers us more flexibility in the future (for example if we wanted to support multiple custom topics we wouldn't need to make any changes to the `p2p` package).

Reply:
I like this idea. Changed it to `PublishTopics` and `SubscribeTopic` (singular) since we only support a single subscribe topic for now.